lib/client/api.go

```diff
@@ -1544,8 +1551,21 @@ func (tc *TeleportClient) Join(ctx context.Context, namespace string, sessionID
 	// Start forwarding ports if configured.
 	tc.startPortForwarding(ctx, nc)
 
+	presenceCtx, presenceCancel := context.WithCancel(ctx)
+	defer presenceCancel()
+
+	var beforeStart func(io.Writer)
+	if mode == types.SessionModeratorMode {
+		beforeStart = func(out io.Writer) {
+			nc.OnMFA = func() {
+				runPresenceTask(presenceCtx, out, site, tc, string(session.ID))
+			}
+		}
+	}
```

Comment:
This doesn't imply MFA, this merely registers a callback that's triggered by an MFA request from the session.
This part of the code doesn't yet know if the session will require MFA, yet it has the various bits of data we need to set up the presence task.

As a precaution, this sets up a callback that will start a presence task if ran. That callback is hooked up to a handler for the `x-teleport-mfa-presence` out of bands SSH event. This event is sent ASAP from the server after a client joins to inform it of the MFA requirement. We don't need to wait for the first MFA verification before we join since an MFA verification is needed to issue the certs to join the session.